samples/net/sockets/http_server/prj.conf

```diff
@@ -69,14 +69,21 @@ CONFIG_NET_SOCKETS_SOCKOPT_TLS=y
 CONFIG_NET_SOCKETS_TLS_MAX_CONTEXTS=6
 CONFIG_TLS_CREDENTIALS=y
 CONFIG_TLS_MAX_CREDENTIALS_NUMBER=5
-CONFIG_MBEDTLS_ECDH_C=y
-CONFIG_MBEDTLS_ECDSA_C=y
-CONFIG_MBEDTLS_ECP_C=y
-CONFIG_MBEDTLS_ECP_DP_SECP256R1_ENABLED=y
-CONFIG_MBEDTLS_KEY_EXCHANGE_RSA_ENABLED=n
```

Comment:
Should we leave these and have a separate overlay config file for PSA options so that the legacy compilation works as before?
And then have separate section that tests the non-PSA and PSA compilations in `sample.yaml` file?
So have something like
```
sample.net.sockets.http.server: {}
sample.net.sockets.http.server.no_psa:
    filter: not CONFIG_BUILD_WITH_TFM
    extra_configs:
      - CONFIG_MBEDTLS_PSA_CRYPTO_C=y
      here setting the psa overlay
  sample.net.sockets.http.server.psa:
   filter: CONFIG_BUILD_WITH_TFM
      here setting the psa overlay
```
WDYT?